### PR TITLE
magnetic field of straight line

### PR DESCRIPTION
**Pull request by Priyanka-43**
In this pull request, I have written documentation part of Magnetic Field of wire and try to elaborate with figure as well.

### CODE REVIEW

**Review by kylemasc917 — CHANGES_REQUESTED** [review]
This documentation file is complete and well written, all equations are legible latex code.
Just a few things need to be addressed.
1. A description of the materials in the pull request needs to be added.
2. Are the files for the rest of the homework assignment going to be added in a later pull request?
                    -If so, this can be merged once 1. is addressed.
                    -If not, then those files need to be added as soon as possible.

**Comment by laserlab** [comment]
> This documentation file is complete and well written, all equations are legible latex code. Just a few things need to be addressed.
> 
> 1. A description of the materials in the pull request needs to be added.
> 2. Are the files for the rest of the homework assignment going to be added in a later pull request?
>    -If so, this can be merged once 1. is addressed.
>    -If not, then those files need to be added as soon as possible.

Agreed. 
@kylemasc917 do you want to also review #46 since there is actual code to review? You don’t have to you’ve already done your duty.

**Comment by Priyanka-43** [comment]
@kylemasc917 
Hi, Can you please explain little more for your first comment? 

And i have made another Pull request for Code.

**Comment by kylemasc917** [comment]
Yeah. The Pull request currently has no description provided (the box at the top is blank). In order for this to be merged according to the homework assignment there needs to be a brief description of the contents of the pull request. You can do this by pressing the three dots in the top right of the first box, pressing edit, and adding a description.

**Comment by laserlab** [comment]
> @kylemasc917 Hi, Can you please explain little more for your first comment?
> 
> And i have made another Pull request for Code.

Can you link that PR here? There is no other authored by you in the open PR list in this repository

**Comment by laserlab** [comment]
@Priyanka-43 As per homework description only one PR is to be made. If you want to code count to your homework you have to add the files to this PR here NOW.

**Comment by Priyanka-43** [comment]
Ok, I have added my code part to the same pull request.

**Comment by laserlab** [comment]
> Ok, I have added my code part to the same pull request.

@kylemasc917 Would you mind reviewing the code as well?

**Review by kylemasc917 — CHANGES_REQUESTED** [review]
Above issues have been resolved and code runs well and is extremely readable.

Only further changes are that in order to complete the homework assignment there needs to be a unit test module that accompanies the code which is not currently present.

**Comment by Priyanka-43** [comment]
@kylemasc917 
Hello, I wanted to ask about the unit test part. Shall we used pytest an unit test libraries to make sure it will be running or if it is already giving the results so I am done with the unit test part?

**Comment by kylemasc917** [comment]
Yes you need to write a unit test module even if the code is running properly as a check like constructing a proof for a problem even if you got the expected value.

**Comment by kylemasc917** [comment]
A few things. This looks like it is written in Jupyter notebook which will not work as was mentioned in class on Friday I believe, and you never import the function from your code module into the unit test, so it isn't testing anything. If I'm misreading this code feel free to tell me so but those are my observations.

File: Code_MagneticFieldOfWire.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+#Python code for Magnetic field of wire
+import math
```
Comment (laserlab) [inline Code_MagneticFieldOfWire.py:2]: all import should be at the top of the file or at least grouped together

File: Code_MagneticFieldOfWire.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+plt.show()
+
+
+
+
+
+#Python code for Magnetic field of wire Using Generator function
+
+
+import numpy as np
```
Comment (laserlab) [inline Code_MagneticFieldOfWire.py:69]: multiple import of the same module

File: Code_MagneticFieldOfWire.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import matplotlib.pyplot as plt
```
Comment (laserlab) [inline Code_MagneticFieldOfWire.py:70]: see above and also first comment

File: Code_MagneticFieldOfWire.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+#Python code for Magnetic field of wire Using Lambda function
+import numpy as np
+import matplotlib.pyplot as plt
+
+# Constants
+mu_0 = 4 * np.pi * 10**-7  # Permeability of free space (T m/A)
+
+# Parameters
+current = 10  # Current in amperes
```
Comment (laserlab) [inline Code_MagneticFieldOfWire.py:41]: reuse of variable without prior use

File: Code_MagneticFieldOfWire.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+# Use np.array to create an array of distances
+distances = np.array([0.01, 0.1, 0.2, 0.3, 0.4, 0.5, 100])  # Distances from 0.01 m to 100 m
+
+# Lambda function to calculate magnetic field
+magnetic_field = lambda I, r: (mu_0 * I) / (2 * np.pi * r)
```
Comment (laserlab) [inline Code_MagneticFieldOfWire.py:46]: Strongly discourage use of lambda function as discussed in class

File: Code_MagneticFieldOfWire.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+# Calculate magnetic field values using map with the lambda function
+fields = list(map(lambda r: magnetic_field(current, r), distances))
+
+# Plotting
+plt.figure(figsize=(10, 6))
```
Comment (laserlab) [inline Code_MagneticFieldOfWire.py:52]: that is not intended behavior when importing a module. Give it a try and import your module and see if that is what you wanted it to do.

File: Code_MagneticFieldOfWire.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+plt.plot(distances, fields)
+plt.title('Magnetic Field Around a Straight Current-Carrying Wire')
+plt.xlabel('Distance from the Wire (m)')
+plt.ylabel('Magnetic Field (T)')
+plt.grid(True)
+plt.ylim(0, max(fields) * 1.1)
+plt.xlim(0, 0.5)  # Limit x-axis to 0.5 to focus on the initial part of the curve
+plt.show()
```
Comment (laserlab) [inline Code_MagneticFieldOfWire.py:60]: plots in particular with the show function are much better suited for a jupyter notebook and should not be in a module in most cases.

File: Code_MagneticFieldOfWire.py [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+import numpy as np
+import matplotlib.pyplot as plt
+
+# Constants
+mu_0 = 4 * np.pi * 10**-7  # Permeability of free space (T m/A)
+
+def magnetic_field_generator(current, distances):
+    """Generator function to yield magnetic field values for given distances."""
+    for r in distances:
+        yield (mu_0 * current) / (2 * np.pi * r)
```
Comment (laserlab) [inline Code_MagneticFieldOfWire.py:78]: np.pi and math.pi used in the same module for unclear reasons

File: MagneticFieldofStraightwire.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
+# Documentation
+
+## Magnetic Field of Straight wire
+
+It is understood that a moving charge or current generates a magnetic field. A long straight wire carrying a current I is 
+shown in diagram below. The presence of current in the wire creates a magnetic field around it which consist of concentric
+circles. 
+![Screenshot 2024-10-13 164423](https://github.com/user-attachments/assets/25840c22-e22b-402f-86fb-426b548127b3)
```
Comment (laserlab) [inline MagneticFieldofStraightwire.md:8]: If you do not have the copyright to this image please remove. If you have it needs a reference.

**Review by laserlab — CHANGES_REQUESTED** [review]
See inline comments. Also this does not use your electrodynamic function inside a function.

File: MagneticFieldofStraightwire.md (hunk 9 above)
Comment (Priyanka-43) [reply MagneticFieldofStraightwire.md:8]: Hello, I have made this picture by myself.

File: Code_MagneticFieldOfWire.py (hunk 8 above)
Comment (Priyanka-43) [reply Code_MagneticFieldOfWire.py:78]: I have changes in math.pi

File: Code_MagneticFieldOfWire.py (hunk 7 above)
Comment (Priyanka-43) [reply Code_MagneticFieldOfWire.py:60]: Ok, Professor, I  will take care of that.

File: Code_MagneticFieldOfWire.py (hunk 6 above)
Comment (Priyanka-43) [reply Code_MagneticFieldOfWire.py:52]: Sure.

File: Code_MagneticFieldOfWire.py (hunk 5 above)
Comment (Priyanka-43) [reply Code_MagneticFieldOfWire.py:46]: Sure.

File: Code_MagneticFieldOfWire.py (hunk 4 above)
Comment (Priyanka-43) [reply Code_MagneticFieldOfWire.py:41]: I have changed into one time use of variable.

File: Code_MagneticFieldOfWire.py (hunk 3 above)
Comment (Priyanka-43) [reply Code_MagneticFieldOfWire.py:70]: Yes. Changed.

File: Code_MagneticFieldOfWire.py (hunk 2 above)
Comment (Priyanka-43) [reply Code_MagneticFieldOfWire.py:69]: Changed into one.

File: Code_MagneticFieldOfWire.py (hunk 5 above)
Comment (laserlab) [reply Code_MagneticFieldOfWire.py:47]: Same as before

**Review by laserlab — CHANGES_REQUESTED** [review]
- plenty of issues left no merge

**Comment by Priyanka-43** [comment]
Hello Professor, I have tried to resolve. I have done all import in starting of the code. i have deleted the parameters(variables) which I have used more than one. Try to take one module math.pi or np.pi.

**Pull request closed without merging**